api/src/services/stats.service/engagement.js

```diff
@@ -628,22 +629,27 @@ async function getYoungStartPhase2InTime(startDate, endDate, user) {
 
   const response2 = await esClient.search({ index: "contract", body: body2 });
 
-  let value = response2.body.aggregations.group_by_youngId.buckets;
+  const buckets = response2.body.aggregations.group_by_youngId.buckets;
+  const value = [];
 
-  value = value.filter((bucket) => {
+  //@todo : TO TEST @C2Chandelier
+  for await (const bucket of buckets) {
     const minYoungContractValidationDate = new Date(bucket.minYoungContractValidationDate.value);
     const correspondingYoung = youngsCohort.find((young) => young._id === bucket.key);
 
     //check si la date de validation de contract est moins d'un an après la date de validation de phase 1 du jeune
     if (correspondingYoung) {
       const oneYearInMilliseconds = 365 * 24 * 60 * 60 * 1000;
-      const cohortEnd = COHESION_STAY_END[correspondingYoung.cohort];
+      const cohort = await CohortModel.findOne({ name: correspondingYoung.cohort });
+      const cohortEnd = cohort?.dateEnd || END_DATE_PHASE1[correspondingYoung.cohort];
 
```

Comment:
puisqu'on est dans un for pour éviter de faire 100 appel a la DB, on pourrait faire un seul appel en amont avec toutes les cohortes du type :  const cohortArray = [cohort.name, cohort.dateEnd] 
et dans le for : const cohortEnd = cohortArray[correspondingYoung.cohort] || END_PHASE1[correspondingYoung.cohort]
T'en penses quoi ?